src/gui/static/src/app/services/exchange.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Observable } from 'rxjs/Observable';
+import { ExchangeOrder, TradingPair } from '../app.datatypes';
+import { environment } from '../../environments/environment';
+
+@Injectable()
+export class ExchangeService {
+  private readonly API_ENDPOINT = 'https://swaplab.cc/api/v3';
+  private readonly API_KEY = 'w4bxe2tbf9beb72r';
+
+  private lastOrder: ExchangeOrder;
+
+  constructor(
+    private http: HttpClient,
+  ) { }
+
+  tradingPairs(): Observable<TradingPair[]> {
+    return this.post('trading_pairs').map(data => data.result);
+  }
+
+  exchange(pair: string, fromAmount: number, toAddress: string): Observable<ExchangeOrder> {
+    return this.post('orders', { pair, fromAmount, toAddress }).map(data => data.result);
```

Comment:
Should not this function also receive and send the `refundAddress` parameter? Probably that will make things easier in case of any error.